backend/src/config.ts

```diff
@@ -2,24 +2,18 @@ import * as fs from "fs";
 import { z } from "zod";
 import { ChainId } from "./types";
 
-const AnyChainConfig = z.object({
+const PrimaryChainConfig = z.object({
   downloadedArchivePath: z.string().optional(),
-  httpUrl: z.string(),
   accountSeed: z.string(),
   feedId: z.number().refine((number) => {
     return number >= 0;
   }),
-});
-
-export type AnyChainConfig = z.infer<typeof AnyChainConfig>;
-
-const PrimaryChainConfig = AnyChainConfig.extend({
   wsUrl: z.string(),
 });
 
 export type PrimaryChainConfig = z.infer<typeof PrimaryChainConfig>;
 
-const ParachainConfig = AnyChainConfig.extend({
+const ParachainConfig = PrimaryChainConfig.extend({
```

Comment:
I would leave `AnyChainConfig` to make reader aware that parachain is not an extension of primary chain, it is a different data structure that happens to have some fields in common.